### PR TITLE
[Merged by Bors] - chore: restore tfae proofs

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Inseparable.lean [hunk 1]
```diff
@@ -70,38 +70,26 @@ infixl:300 " ⤳ " => Specializes
...
-  --   rintro s ⟨ hy , ho ⟩
-  --   rcases mem_closure_iff . 1 h s ho hy with ⟨ z , hxs , rfl : z = x ⟩
-  --   exact ho.mem_nhds hxs
-  -- tfae_finish
+  tfae_have 1 → 2; exact (pure_le_nhds _).trans
+  tfae_have 2 → 3; exact fun h s hso hy => h (hso.mem_nhds hy)
+  tfae_have 3 → 4; exact fun h s hsc hx => of_not_not fun hy => h (sᶜ) hsc.isOpen_compl hy hx
+  tfae_have 4 → 5; exact fun h => h _ isClosed_closure (subset_closure <| mem_singleton _)
+  tfae_have 6 ↔ 5; exact isClosed_closure.closure_subset_iff.trans singleton_subset_iff
+  tfae_have 5 ↔ 7; · rw [mem_closure_iff_clusterPt, principal_singleton]
```
Comment (semorrison) [inline Mathlib/Topology/Inseparable.lean:86]:
```suggestion
  tfae_have 5 ↔ 7; rw [mem_closure_iff_clusterPt, principal_singleton]
```

Comment (semorrison) [reply Mathlib/Topology/Inseparable.lean:86]: Maybe I don't understand why you wrote it this way. It seems strange to focus for the `rw` proofs but not for the `exact` proofs?

File: Mathlib/Topology/Order/Basic.lean [hunk 2]
```diff
@@ -1590,16 +1591,21 @@ theorem TFAE_mem_nhdsWithin_Ioi {a b : α} (hab : a < b) (s : Set α) :
...
-    rcases exists_Ico_subset_of_mem_nhds' h hab with ⟨u, au, hu⟩
-    exact ⟨u, au, fun x hx => hu (Ioo_subset_Ico_self hx) hx.1⟩
-  · rintro ⟨u, hu, hs⟩
-    exact ⟨u, hu.1, hs⟩
-  · rintro ⟨u, hu, hs⟩
-    exact mem_of_superset (Ioo_mem_nhdsWithin_Ioi' hu) hs
+  tfae_have 1 ↔ 2;
+  · rw [nhdsWithin_Ioc_eq_nhdsWithin_Ioi hab]
+  tfae_have 1 ↔ 3; · rw [nhdsWithin_Ioo_eq_nhdsWithin_Ioi hab]
+  tfae_have 4 → 5; exact fun ⟨u, umem, hu⟩ => ⟨u, umem.1, hu⟩
```
Comment (semorrison) [inline Mathlib/Topology/Order/Basic.lean:1597]: Similarly here?

**Comment by semorrison** [comment]
bors d+

File: Mathlib/Topology/Inseparable.lean (hunk 1 above)
Comment (digama0) [reply Mathlib/Topology/Inseparable.lean:86]: `exact` always closes the goal, `rw` doesn't necessarily. I think it is better with the `· rw`

Comment (Ruben-VandeVelde) [reply Mathlib/Topology/Inseparable.lean]: This was copied straight from mathport, but I regularized the formatting a bit.

**Pull request closed without merging**